### PR TITLE
feat(pt_expt): implement DeepSpin model in pt_expt backend 

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
## Summary                                                                                                                                                                                                                                                   
                                         
  - Implement `SpinModel` and `SpinEnergyModel` in the pt_expt backend, supporting spin degrees of freedom for magnetic systems                                                                                                                                
  - Make dpmodel `SpinModel` array-API compatible so the same code works across numpy/torch/jax backends
  - Add spin virial correction (`coord_corr_for_virial`) to dpmodel and pt_expt, matching the pt backend                                                                                                                                                       
  - Fix `get_spin_model` in dpmodel to not mutate the caller's input data dict (pt backend already used `deepcopy`)

  ## Changes

  ### dpmodel (`deepmd/dpmodel/model/`)

  - `spin_model.py`: Replace all `np.*` operations with `array_api_compat` equivalents (`xp.concat`, `xp.where`, `xp.zeros` with `device=`, slicing instead of `xp.split`). Add `compute_or_load_stat` and virial correction support via
  `coord_corr_for_virial` / `extended_coord_corr`.
  - `make_model.py`: Thread `coord_corr_for_virial` through `call_common` → `model_call_from_call_lower` (extends to ghost atoms via mapping) → `call_common_lower` → `forward_common_atomic`.
  - `model.py`: Add `copy.deepcopy(data)` in `get_spin_model` to prevent in-place mutation of input dict.

  ### pt_expt (`deepmd/pt_expt/model/`)

  - `spin_model.py` (new): `@torch_module` wrapper inheriting from dpmodel `SpinModel`.
  - `spin_ener_model.py` (new): `SpinEnergyModel` with `forward()` / `forward_lower()` / `forward_lower_exportable()` providing user-facing output translation.
  - `make_model.py`, `transform_output.py`: Accept `extended_coord_corr` for virial correction.

  ### Tests

  - `test_spin_ener_model.py` (new): Unit tests for output keys/shapes, serialize/deserialize round-trip, dpmodel consistency, force finite-difference, virial finite-difference, and `torch.export` exportability.
  - `test_spin_ener.py`: Cross-backend consistency tests for `call`/`call_lower`, `compute_or_load_stat`, and load-from-file. Virial output now compared across pt and pt_expt.

  ## Test plan

  - [x] `python -m pytest source/tests/pt_expt/model/ -v` — all 28 tests pass
  - [x] `python -m pytest source/tests/consistent/model/test_spin_ener.py -v` — all 12 tests pass (18 skipped for uninstalled backends)
  - [x] Force and virial verified by finite-difference tests
  - [x] `torch.export.export` verified on `forward_lower_exportable`
  - [x] `compute_or_load_stat` load-from-file verified across dp/pt/pt_expt


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added SpinEnergyModel with exportable lower-level forward, energy/force/virial outputs, and compute_or_load_stat preprocessing.
  * Optional virial coordinate-correction can be supplied and is propagated through forward paths.

* **Bug Fixes**
  * Prevented in-place mutation of input data during model preparation.

* **Tests**
  * Expanded tests for exportable workflows, force/virial validation, multi-backend (including PT_EXPT) and array‑API strict modes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: deepmd/pt_expt/model/spin_ener_model.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        model_predict["atom_energy"] = model_ret["energy"]
+        model_predict["energy"] = model_ret["energy_redu"]
+        model_predict["mask_mag"] = model_ret["mask_mag"]
+        if self.backbone_model.do_grad_r("energy"):
+            model_predict["force"] = model_ret["energy_derv_r"].squeeze(-2)
+            model_predict["force_mag"] = model_ret["energy_derv_r_mag"].squeeze(-2)
+        if self.backbone_model.do_grad_c("energy"):
+            model_predict["virial"] = model_ret["energy_derv_c_redu"].squeeze(-2)
+            if do_atomic_virial:
+                model_predict["atom_virial"] = model_ret["energy_derv_c"].squeeze(-3)
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/pt_expt/model/spin_ener_model.py]: fixed by 587392fa

**Review by iProzd — CHANGES_REQUESTED** [review]
1. Some new api with PR #5281 need to be fixed.
2. Training & loss of spin model in another PR?

**Review by iProzd — APPROVED** [review]
